controllers/breed-controller_test.go

```diff
@@ -95,5 +99,206 @@ func TestCreateBreed(t *testing.T) {
 
 	testService := services.NewBreedService(breedRepoMock)
 	testController := NewBreedController(breedServMock)
+	testService.CreateBreed(dogBreed)
+
+	resp := httptest.NewRecorder()
+	handler := http.HandlerFunc(testController.Create)
+	handler.ServeHTTP(resp, req)
+
+	status := resp.Code
+	if status != http.StatusOK {
+		t.Errorf(err.Error(), "expecting 200 code got: %v", status)
+	}
+
+	var respBody entities.DogBreed
+	err = json.NewDecoder(resp.Body).Decode(&respBody)
+	if err != nil {
+		t.Errorf(err.Error(), "error on body response decoding")
+	}
+
+	breedServMock.AssertExpectations(t)
+	breedRepoMock.AssertExpectations(t)
+
+	assert.Equal(t, resp.Code, 200)
+	assert.Equal(t, respBody.Grooming, dogBreed.ID)
+	assert.Equal(t, respBody.BreedImg, dogBreed.BreedImg)
+	assert.Equal(t, respBody.Energy, dogBreed.Energy)
+	assert.Equal(t, respBody.GoodWithDogs, dogBreed.GoodWithDogs)
+	assert.Equal(t, respBody.GoodWithKids, dogBreed.GoodWithKids)
+	assert.Equal(t, respBody.Grooming, dogBreed.Grooming)
+	assert.Equal(t, respBody.Shedding, dogBreed.Shedding)
+
+}
+
+func TestGetAllBreeds(t *testing.T) {
+
+	breedServMock := new(breedServMock)
+	breedRepoMock := new(breedRepoMock)
+
+	dogBreed := entities.BuildDogBreed("1", "x", 1, 1, 1, 1, 1, 1, 1)
```

Comment:
Um método quando recebe muitos parameter como este, precisa ser refatorado, porque fica muito difícil de entender o que está acontecendo, o que significa este tanto de 1 e x.
Vc pode criar uma struct para receber como parametro, pode ser um DogBreedRequest ou um DogBreedDTO. Aí neste caso nem precisaria deste builder, basta lá no service receber esta struct DTO.
Ou se quiser mander este builder, tem outra forma de fazer como este: 
https://devcharmander.medium.com/design-patterns-in-golang-the-builder-dac468a71194

Reply:
Tá. Cada entidade do meu dominio tem um construtor, que me ajudou nos testes quando eu tinha que instanciar alguma entidade pra testar, esse BuildDogBreed é um desses contrutores, ele recebe o id da breed, o nome da breed, e os seus atributos que sao good with dogs, good with kids, shedding, grooming, energy, etc. 
Realmente fica meio feio de se visualizar, vou olhar o artigo que tu mandou!